Mathlib/Topology/Order/UpperLowerSetTopology.lean

```diff
@@ -0,0 +1,475 @@
+  intro s hs
+  rw [IsOpen_iff_IsUpperSet]
+  apply IsUpperSet.preimage _ hf
+  apply UpperTopology.isUpperSet_of_isOpen hs
+
+lemma UpperSetLEUpper {t₁ : TopologicalSpace α} [@UpperSetTopology α t₁ _]
+    {t₂ : TopologicalSpace α} [@UpperTopology α t₂ _] : t₁ ≤ t₂ := fun s hs => by
+  rw [@IsOpen_iff_IsUpperSet α _ t₁]
+  exact UpperTopology.isUpperSet_of_isOpen hs
+
+end maps
+
+end UpperSetTopology
+
+namespace LowerSetTopology
+
+section Preorder
+
+variable (α)
+variable [Preorder α] [TopologicalSpace α] [LowerSetTopology α] {s : Set α}
+
+lemma topology_eq : ‹_› = lowerSetTopology' α := topology_eq_lowerSetTopology
+
+variable {α}
+
+instance instUpperSetTopologyDual [Preorder α] [TopologicalSpace α] [LowerSetTopology α] :
+    UpperSetTopology (αᵒᵈ) where
+  topology_eq_upperSetTopology := by
+    refine topologicalSpace_eq ?_
+    rw [(LowerSetTopology.topology_eq (α))]
+
+/-- If `α` is equipped with the lower set topology, then it is homeomorphic to
+`WithLowerSetTopology α`.
+-/
+def withLowerSetTopologyHomeomorph : WithLowerSetTopology α ≃ₜ α :=
+  WithLowerSetTopology.ofLowerSet.toHomeomorphOfInducing ⟨by erw [topology_eq α, induced_id]; rfl⟩
+
+lemma IsOpen_iff_IsLowerSet : IsOpen s ↔ IsLowerSet s := by
+  rw [topology_eq α]
+  rfl
+
+-- Alexandrov property, set formulation
+theorem IsOpen_sInter {S : Set (Set α)} (hf : ∀ s ∈ S, IsOpen s) : IsOpen (⋂₀ S) :=
+  UpperSetTopology.IsOpen_sInter (α := αᵒᵈ) (fun s a ↦ hf s a)
+
+-- Alexandrov property, index formulation
+theorem isOpen_iInter {f : ι → Set α} (hf : ∀ i, IsOpen (f i)) : IsOpen (⋂ i, f i) :=
+  UpperSetTopology.isOpen_iInter (α := αᵒᵈ) hf
+
+lemma isClosed_iff_isUpper {s : Set α} : IsClosed s ↔ (IsUpperSet s) := by
+  rw [← isOpen_compl_iff, IsOpen_iff_IsLowerSet, isUpperSet_compl.symm, compl_compl]
+
+lemma isClosed_isUpper {s : Set α} : IsClosed s → IsUpperSet s := fun h =>
+  (isClosed_iff_isUpper.mp h)
+
+lemma closure_eq_upperClosure {s : Set α} : closure s = upperClosure s :=
+  UpperSetTopology.closure_eq_lowerClosure (α := αᵒᵈ)
+/--
+The closure of a singleton `{a}` in the lower set topology is the right-closed left-infinite
+interval (-∞,a].
+-/
+@[simp] lemma closure_singleton {a : α} : closure {a} = Ici a := by
+  rw [closure_eq_upperClosure, upperClosure_singleton]
+  rfl
+
+end Preorder
+
+section maps
+
+variable [Preorder α] [Preorder β]
+
+open Topology
+
+protected lemma monotone_iff_continuous [TopologicalSpace α] [LowerSetTopology α]
+    [TopologicalSpace β] [LowerSetTopology β] {f : α → β} :
+    Monotone f ↔ Continuous f := by
+  constructor
+  · intro hf
+    simp_rw [continuous_def, IsOpen_iff_IsLowerSet]
+    exact fun _ hs ↦ IsLowerSet.preimage hs hf
+  · intro hf a b hab
+    rw [← mem_Ici, ← closure_singleton, ← mem_preimage]
+    apply Continuous.closure_preimage_subset hf
+    rw [← mem_Ici, ← closure_singleton] at hab
+    apply mem_of_mem_of_subset hab
+    apply closure_mono
+    rw [singleton_subset_iff, mem_preimage, mem_singleton_iff]
```

Comment:
Did you try getting this proof by duality? For the shorter one above there's no point since it's really short already, but for this one and the ones below it would be really nice.

Reply:
I've done this now. Needed an extra lemma: #6157